plutus-contract/src/Plutus/Contract/StateMachine/ThreadToken.hs

```diff
@@ -81,6 +81,31 @@ checkThreadTokenInner :: CurrencySymbol -> ValidatorHash -> Value -> Integer ->
 checkThreadTokenInner currency (ValidatorHash vHash) vl i =
     Value.valueOf vl currency (TokenName vHash) == i
 
+-- | Check exactly `n` thread tokens and no other tokens with the given
+-- @CurrencySymbol@ are in the given @Value@.
```

Comment:
@michaelpj You [suggested in the previous PR](https://github.com/input-output-hk/plutus/pull/4105#discussion_r729679862) that this might be too strict. Are you saying that the change is not necessary? My intention with this PR was specifically to disallow minting other tokens of the same CurrencySymbol. My reasoning was that the intention of the ThreadToken is for there to only be one unique token. The CurrencySymbol incorporates a `TxOutRef` used to anchor the NFT. In my mind we shouldn't allow creation of other TokenNames not intended by the StateMachine implementation.